File: python_sample/payloads/upload_file.py
import json
import requests

type_fields = {"name": str, "file": str, "rename_if_exists": bool}


class UploadFile:
    name = ""
    file = ""
    rename_if_exists = False

    def __init__(self, *args, **kwargs):
        mandatory_fields = ["name", "file"]

        for key, val in kwargs.items():
            setattr(self, key, val)
            if key in mandatory_fields:
                mandatory_fields.remove(key)

        if mandatory_fields:
            raise Exception("{} are missing".format(','.join(mandatory_fields)))

    def __setattr__(self, key, value):
        if not isinstance(value, type_fields.get(key)):
            raise Exception("{} should be a {}".format(key, type_fields.get(key)))
        super().__setattr__(key, value)

    def upload_file(self, base_headers, base_url):
        """
        Used to create a signature request on a template
        :param base_url: string, Base URL to be pre fixed to all API
        :param base_headers: dictionary, with all the necessary tokes for transaction
        :return: tuple: json response, status code
        """
        try:
            url = base_url + "v1/files/original/"

            headers = dict(base_headers)
            headers["Content-Type"] = "multipart/form-data"

            payload = self.to_dict()

            with open(payload['file'], 'rb') as file:
                del payload['file']
                resp = requests.post(url=url, headers=headers, json=payload, files={"file": file})
            return resp.json(), resp.status_code

        except Exception as e:
            return e, 400

    def __str__(self):
        resp = dict()
        resp["name"] = self.name
        resp["file"] = self.file
        resp["rename_if_exists"] = self.rename_if_exists
        return json.dumps(resp)

    def to_dict(self):
        resp = dict()
        resp["name"] = self.name
        resp["file"] = self.file
        resp["rename_if_exists"] = self.rename_if_exists
        return resp

File: v1/node_sample/payload/UploadFile.js
var unirest = require('unirest');

class UploadFile {
	constructor(name, file, rename_if_exists=false)
	{
		this.name = name;
		this.file = file;
		this.rename_if_exists = rename_if_exists;
	}

	to_json()
	{
		return {"name": this.name, "rename_if_exists": this.rename_if_exists}
	}

	upload_file = function (base_url, api_token) {
        var fs = require('fs');
        var readStream = fs.createReadStream(this.file);

        var rename_allowed = "0";
        if (this.rename_if_exists == true) {rename_allowed = "1";}

        return new Promise((resolve, reject) => {
        unirest.post(base_url + 'v1/files/original/')
        .headers({'Authorization': api_token, 'Content-Type': 'multipart/form-data'})
        .attach('file', readStream)
        .field("name", this.name)
        .field("rename_if_exists", rename_allowed)
        .end(function (response) {
        if (response.error) {
        return reject(response.body)
        }
        return resolve(response.body);
        });
    })
    }
}

module.exports = UploadFile